src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -1,876 +1,10 @@
-            throws ParamMissingException, ParamInvalidException {
-        ArrayList<Integer> dayOfTheWeek = new ArrayList<>();
-        try {
-            String stringAfterPrefix =
-                    params.split(Command.COMMAND_PREFIX_DAY_OF_THE_WEEK
-                            + Command.COMMAND_PREFIX_DELIMITER, 2)[1];
-            for (int i = 0; i < stringAfterPrefix.length(); i++) {
-                int day = Integer.parseInt(String.valueOf(stringAfterPrefix.charAt(i)));
-                if (day > MONDAY && day < SUNDAY) { //between monday and sunday
-                    if (dayOfTheWeek.contains(day)) {
-                        throw new ParamInvalidException(MESSAGE_ERROR_REPEATED_DAY_OF_THE_WEEK);
-                    }
-                    dayOfTheWeek.add(day);
-                } else {
-                    throw new ParamInvalidException(MESSAGE_ERROR_INVALID_DAY_OF_THE_WEEK);
-                }
-            }
-        } catch (IndexOutOfBoundsException e) {
-            logger.log(Level.WARNING, "Detected empty day input after prefix but day is required!");
-            throw new ParamMissingException(MESSAGE_ERROR_NO_DAY_OF_THE_WEEK);
-        } catch (NumberFormatException e) {
-            logger.log(Level.WARNING, "Detected non-digit calories input");
-            throw new ParamInvalidException(MESSAGE_ERROR_INVALID_DAY_OF_THE_WEEK);
-        }
-        return dayOfTheWeek;
-    }
-
-    private boolean isFutureDate(LocalDate date) {
-        return date.isAfter(LocalDate.now());
-    }
-
-    private int convertItemNumToItemIndex(int itemNum) {
-        return itemNum - 1;
-    }
-
-
-    private int getNumberOfCorrectParamsDetected(String params, String... prefixes) {
-        int count = 0;
-        for (String prefix : prefixes) {
-            if (params.toLowerCase().contains(prefix + Command.COMMAND_PREFIX_DELIMITER)) {
-                count++;
-            }
-        }
-        logger.log(Level.INFO, String.format("no. of corrected params detected: %s", count));
-        return count;
-    }
-
-    private boolean hasExtraDelimiters(String params, int expectedNum) {
-        int numOfDelimiters = 0;
-        for (int i = 0; i < params.length(); i++) {
-            if (params.charAt(i) == Command.COMMAND_PREFIX_DELIMITER.charAt(0)) {
-                numOfDelimiters++;
-            }
-        }
-        return numOfDelimiters > expectedNum;
-    }
-
-    private boolean hasTextFileDelimiter(String input) {
-        return input.contains(FILE_TEXT_DELIMITER);
-    }
-
-    private String correctCommandFormatSuggestions(String... suggestions) {
-        String formattedSuggestions = MESSAGE_ERROR_INVALID_FORMAT + LS;
-        int i = 1;
-        for (String suggestion : suggestions) {
-            formattedSuggestions += i + ". " + suggestion + LS;
-            i += 1;
-        }
-        return formattedSuggestions.stripTrailing();
-    }
+public interface Parser {
```

Comment:
If I'm not wrong, interfaces should end with "-able" but I dont think Parsable is a word so shd be fine right HAHAH